clang/test/Driver/aarch64-fixed-register-global.c

```diff
@@ -0,0 +1,13 @@
+// Check that -ffixed register handled for globals.
+// Regression test for #76426
+// RUN: %clang --target=aarch64-none-gnu -ffixed-x15 -### %s 2>&1 | FileCheck %s
+// CHECK-NOT: fatal error: error in backend: Invalid register name "x15".
```

Comment:
You modified Clang code but this error message is the one of the LLVM backend.
Is it better to check the Clang-side message `error: size of register 'x15' does not match variable size`?

Reply:
Right! I'd keep the bugreport as is for regression test but added a new tests for the reg check.